src/MuTalk-Utilities-Tests/MTMutantOperatorAnalysisTest.class.st

```diff
@@ -0,0 +1,59 @@
+Class {
+	#name : 'MTMutantOperatorAnalysisTest',
+	#superclass : 'TestCase',
+	#instVars : [
+		'operatorAnalysis'
+	],
+	#category : 'MuTalk-Utilities-Tests',
+	#package : 'MuTalk-Utilities-Tests'
+}
+
+{ #category : 'running' }
+MTMutantOperatorAnalysisTest >> setUp [
+
+	super setUp.
+
+	operatorAnalysis := MTMutantOperatorAnalysis forClasses:
+		                    { MTAuxiliarClassForMutantOperatorAnalysis }
+]
+
+{ #category : 'tests' }
+MTMutantOperatorAnalysisTest >> testGetAllOperators [
+
+	| actual expected |
+	actual := (operatorAnalysis operatorsProducingOverXMutants: 0) keys
+		          asSet.
+	expected := (MTMutantOperator contentsAll collect: #species) asSet.
+
+	self assert: actual equals: expected
+]
+
+{ #category : 'tests' }
+MTMutantOperatorAnalysisTest >> testOperatorsProducingOver [
+
+	| actual expected |
+	actual := (operatorAnalysis operatorsProducingOverXMutants: 3) keys
+		          asSet.
+	expected := Set
+		            with: MTLiteralIntegersIncreaseOperator
+		            with: MTLiteralIntegersToZeroOperator
+		            with: MTLiteralIntegersDecreaseOperator.
+
+	self assert: actual equals: expected
+]
+
+{ #category : 'tests' }
+MTMutantOperatorAnalysisTest >> testOperatorsProducingUnder [
+
+	| actual expected |
+	actual := (operatorAnalysis operatorsProducingUnderXMutants: 2) keys
+		          asSet.
```

Comment:
Here it's the same but the other way around.
- the `keys asSet` looks like the same smell (but duplicated code?)
- under is < x or <= x ? :D